### PR TITLE
SickChill: fix broken package and update

### PR DESCRIPTION
**Pull request by BKSteve**
_Motivation:_ Fix ownership on the DSM6 package so SickChill can update successfully without having to ssh and chown the env folder
_Linked issues:_  #4703 and many on SickChill [https://github.com/SickChill/SickChill/discussions/7364](https://github.com/SickChill/SickChill/discussions/7364) [https://github.com/SickChill/SickChill/issues/7316](https://github.com/SickChill/SickChill/issues/7316) \#7234 \#7249 \#7265 \#7293 \#7295 \#7321 \#7327 \#7347 \#7410 \#7411

This is the first fix to get the ownership of the /env folder under sc user rather than root so that updates of SickChill can access and utilize the folder for updated python packages. Therefore it's numbered as 20210329-2 as it uses the same original version of SickChill in the cross environment with only the source location changed so it loads.

Yes this still has the #4538 item 1) as python is referred to using \usr\local\python38. But as this is an intermediate small fix only for DSM6 and in prep for the move to a combined DSM6 and DSM7 package this will be tackled soon.

If others wish to try this version here's a link to the spk. [sickchill_noarch-dsm6_20210329-2.spk](https://drive.google.com/file/d/1zHdhkL1sgJBAGtEMO04QZLMo0uboTxpA/view?usp=sharing)

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

File: spk/sickchill/src/service-setup.sh
Comment (publicarray) [inline spk/sickchill/src/service-setup.sh:1]:
```suggestion
PYTHON_DIR="/var/packages/python38/target"
```

**Review by publicarray — CHANGES_REQUESTED** [review]
Thanks the Installation and upgrade succeeds and fixes the package. Unfortunately running the internal updater breaks the package again. Can this be fixed? It might be something the sickchill developers have to do.

FYI installation logs this error but does not appear to impact the package.

```sh
grep --ignore-case error /var/log/packages/sickchill.log
ERROR: SQLAlchemy-1.3.20-cp37-none-any.whl is not a supported wheel on this platform.
```

File: spk/sickchill/src/service-setup.sh
Comment (publicarray) [reply spk/sickchill/src/service-setup.sh:1]: `/usr/local/` is an old path, and we recommend to switch to the new path.

Comment (BKSteve) [reply spk/sickchill/src/service-setup.sh:1]: noted.

**Comment by miigotu** [comment]
Some changes to SC are related to this, and the next release/push to master from develop fixes updating. But we absolutely need fixes in the Synocommunity package for it to ever work. I wish your team would open up a little on changes to this specific package, because it only effects users of SickChill, and we absolutely know what we need for it to be able to run on syno, reliably. If we aren't creating a security issue or some sort of system wide critical error please just let some changes in. If you want a huge change set at once, let us know. We can do a bunch of small PRs with easy to review changes, or a major PR that makes it work for dsm7 and also fix the installer and updater issues at the same time. Just tell us what we need to do to get something merged, otherwise we need to go back to releasing our own spk's, which I don't want to do really.

**Comment by publicarray** [comment]
I just thought this would be an easy PR to merge and fix the package and would obviously prefer if the updater worked.
But can publish it like this to at least fix the installation of the published package.

I haven't looked at #4743 yet, It looks like it doesn't compile yet.

> If you want a huge change set at once, let us know. We can do a bunch of small PRs with easy to review changes, or a major PR that makes it work for dsm7 and also fix the installer and updater issues at the same time

Ideal PR size is hard to judge. Usually 1 PR per SPK package (can include all of: updates/features/cross packages/very small core framework changes) and 1 PR for large changes needed in the framework. If changes to dsm7 are significant enough it might be best to separate it out to a separate PR.

> But we absolutely need fixes in the Synocommunity package for it to ever work. I wish your team would open up a little on changes to this specific package, because it only effects users of SickChill, and we absolutely know what we need for it to be able to run on syno, reliably. If we aren't creating a security issue or some sort of system wide critical error please just let some changes in.

I'm not sure if I understand, Maybe it helps to explain my/our general process. @hgy59 feel free to correct me

All packages must compile on all architectures that report themselves as supported. (e.g. pass CI [sometimes there is an exemption e.g. when a known broken package courses an error in the CI, this is from when the CI did not exist]). We can help get you there, but our tiny team is stretched pretty thin. Then we look over the code (I think this where everyone has a slightly different view on things e.g. I don't have much knowledge of Python with the framework, so I usually don't review them, but I know DSM requirements well), Lastly we build package and do some tests (in my case in a virtual environment) of [basic functionally](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks) and look in log files for errors. Then we merge and publish (master is stable and is what is published)

IMHO this PR failed the update test which I think is part of basic functionality testing. The package is otherwise functional. But since this package is already broken I'm happy to merge this.

As to my question then 

>  Can this be fixed?

I Just wanted to know what the plan with the updater is. The answer would determine if there are any changes I would recommend adding to this PR. If for instance it would not work in any way we would have to notify users that the updater is not working. So Should I wait until `develop` is merged to then publish this?

File: spk/sickchill/src/service-setup.sh
Comment (publicarray) [reply spk/sickchill/src/service-setup.sh]: Sorry I missed this (by resolving your message was hidden). Thanks for the information.

File: spk/sickchill/src/service-setup.sh
Comment (publicarray) [inline spk/sickchill/src/service-setup.sh:65]: FYI `>> ${INST_LOG} 2>&1`  /  `>> ${INST_LOG}` is no longer necessary. I recommend removing it

**Comment by publicarray** [comment]
FYI After installing the package about 5 times in about half an hour I got this message.

`2021-09-08 23:32:09 Unable to setup GitHub properly. GitHub will not be available. Error: HTTPSConnectionPool(host='api.github.com', port=443): Max retries exceeded with url: /orgs/SickChill (Caused by ConnectTimeoutError(, 'Connection to api.github.com timed out. (connect timeout=15)'))`

What could course this? Is there a way to be more respectful of GitHub resources?

**Review by publicarray — APPROVED** [review]

**Comment by publicarray** [comment]
Never mind. I included the patch here

**Comment by publicarray** [comment]
@hgy59 I'm sorry but can you the dsm7 version. Using the CI to publish no-arch packages was not a good idea of mine...

**Comment by BKSteve** [comment]
DSM 6 spk while we wait for sc release.
[sickchill_noarch-dsm6_20210729-2.spk](https://drive.google.com/file/d/1Qp2h5xTQJc99Y7P7YLcniFGXqx89BRnu/view?usp=sharing)

**Comment by publicarray** [comment]
Please link to https://packages.synocommunity.com/sabnzbd/49/sabnzbd.v49.f15047%5Bapollolake-avoton-braswell-broadwell-broadwellnk-bromolow-cedarview-denverton-dockerx64-geminilake-grantley-purley-kvmx64-v1000-x86-x86_64%5D.spk

**Comment by publicarray** [comment]
You can use this link too. 

https://packages.synocommunity.com/sickchill/2/sickchill.v2.f7321%5Bnoarch%5D.spk

**Comment by BKSteve** [comment]
Hi guys, the attempt to put the patch on the file `init_helpers.py` hasn't been successful people are getting failures.

@miigotu can you push the fix to master please.

**Comment by publicarray** [comment]
> Hi guys, the attempt to put the patch on the file init_helpers.py hasn't been successful people are getting failures.
They used the linked file in post above.

Please be specific what fails? I know the patch does not fix the updater, but it did fix the build which I assumed this was about

**Comment by BKSteve** [comment]
Yes, the final version on `init_helpers.py` installed is not amended with the information in the `01-fix-python3.8.patch` file.

**Comment by miigotu** [comment]
As soon as I can get to my laptop, I will make a release of SC, but this patch should apply on a new install. I'm only concerned if patching here breaks the updater in SC. Ideally, we should remove the patch and update the commit hash included in the package. (This is why it was not pinned to a specific release originally, it was from the master branch (which is always the latest release) just like sickbeard-custom.) Any errors were easily fixed on SC end without a new PR here.

**Comment by miigotu** [comment]
Thanks for merging this though, it's a step in the right direction and what is most important is to see progress. Otherwise it just gets frustrating having perpetual requested changes on PRs that actually work.

**Comment by BKSteve** [comment]
The patch inside the package doesn't seem to have any adverse effects, I removed the patch and built the package. Installed it, swapped the init_helpers file and it starts and with some basic testing functions.
Just hanging for the new package to drop.

I also built DSM7 version and did a test and installs.
So once the updated SickChill is available I'll do a full test on both 6 and 7 and create a new pull request linked to this one.
As we've already moved to 20210729 I don't believe git is a requirement anymore, so I'll test that too.

**Comment by miigotu** [comment]
> The patch inside the package doesn't seem to have any adverse effects, I removed the patch and built the package. Installed it, swapped the init_helpers file and it starts and with some basic testing functions.
> Just hanging for the new package to drop.
> 
> I also built DSM7 version and did a test and installs.
> So once the updated SickChill is available I'll do a full test on both 6 and 7 and create a new pull request linked to this one.
> As we've already moved to 20210729 I don't believe git is a requirement anymore, so I'll test that too.

Yeah I've been working on it, there were some issues with current develop and I'm not about to be rushed into another disaster release to partially help only one ecosystem. Maybe tomorrow it will be sorted out for testing.

**Comment by miigotu** [comment]
> The patch inside the package doesn't seem to have any adverse effects, I removed the patch and built the package. Installed it, swapped the init_helpers file and it starts and with some basic testing functions.
> Just hanging for the new package to drop.
> 
> I also built DSM7 version and did a test and installs.
> So once the updated SickChill is available I'll do a full test on both 6 and 7 and create a new pull request linked to this one.
> As we've already moved to 20210729 I don't believe git is a requirement anymore, so I'll test that too.

Yeah I've been working on it, there were some issues with current develop and I'm not about to be rushed into another disaster release to partially help only one ecosystem. Maybe tomorrow it will be sorted out for testing.

**Comment by BKSteve** [comment]
> 
> 
> The patch inside the package doesn't seem to have any adverse effects, I removed the patch and built the package. Installed it, swapped the init_helpers file and it starts and with some basic testing functions.
> Just hanging for the new package to drop.
> 
> I also built DSM7 version and did a test and installs.
> So once the updated SickChill is available I'll do a full test on both 6 and 7 and create a new pull request linked to this one.
> As we've already moved to 20210729 I don't believe git is a requirement anymore, so I'll test that too.

Based on the #4856 issue thread it's obvious the online package is duff.
Please advise what actions are required to rectify the failing online package?

**Comment by publicarray** [comment]
I've already unpublished it yesterday, so It already stops showing up as an update in the package center. The website will still show it though

**Comment by BKSteve** [comment]
The 20210329-1 version also doesn't load now due to [https://github.com/SynoCommunity/spksrc/issues/4861](https://github.com/SynoCommunity/spksrc/issues/4861)

**Comment by BKSteve** [comment]
As I'm working on a fix for all the issues that have become apparent with moving to Python 38 package and particularly for ARM based architectures I have the code and requirements for 38 but for some reason
The install on a DS216j gives this error
`ERROR: tornado-6.1-cp37-none-any.whl is not a supported wheel on this platform.`
which implies that it's building a 3.7 wheel, so how can I force the build environment to build a cp38 wheel or fix this error on ARM based machines.
grateful for any help @publicarray and others

**Comment by BKSteve** [comment]
Reverted version 20210329-1 doesn't seem load same requirements.
[https://github.com/SynoCommunity/spksrc/issues/4861#issuecomment-921519483](https://github.com/SynoCommunity/spksrc/issues/4861#issuecomment-921519483)

**Pull request merged**
